src/insert.rs

```diff
@@ -255,15 +277,22 @@ impl<T> Insert<T> {
     }
 
     async fn wait_handle(&mut self) -> Result<()> {
-        match timeout!(self, end_timeout, &mut self.handle) {
-            Some(Ok(res)) => res,
-            Some(Err(err)) if err.is_panic() => panic::resume_unwind(err.into_panic()),
-            Some(Err(err)) => Err(Error::Custom(format!("unexpected error: {err}"))),
-            None => {
-                // We can do nothing useful here, so just shut down the background task.
-                self.handle.abort();
-                Err(Error::TimedOut)
+        match self.state.handle() {
+            Some(handle) => {
+                let result = match timeout!(self, end_timeout, &mut *handle) {
+                    Some(Ok(res)) => res,
+                    Some(Err(err)) if err.is_panic() => panic::resume_unwind(err.into_panic()),
+                    Some(Err(err)) => Err(Error::Custom(format!("unexpected error: {err}"))),
+                    None => {
+                        // We can do nothing useful here, so just shut down the background task.
+                        handle.abort();
+                        Err(Error::TimedOut)
+                    }
+                };
+                self.state = InsertState::Completed;
```

Comment:
If panic happens, this isn't executed. In general, it should never happen, although.